src/dune_rules/lib.ml

```diff
@@ -1084,7 +1098,10 @@ end = struct
   module Input = struct
     type t = Lib_name.t * Path.t Lib_info.t * string option
 
-    let equal (x, _, _) (y, _, _) = Lib_name.equal x y
+    let equal (lib_name, info, _) (lib_name', info', _) =
+      Lib_name.equal lib_name lib_name' && Lib_info.equal info info'
```

Comment:
If we don't properly distinguish between libraries with different info, the paths will be broken (when getting the info for library `foo` in folder `b`, we would get the lib `foo` in folder `a`).